runner/runners/queue.go

```diff
@@ -60,13 +73,24 @@ func NewSingleRunner(
 // QueueController maintains a queue of commands to run (up to capacity).
 type QueueController struct {
 	inv           *Invoker
+	filer         snapshot.Filer
 	statusManager *StatusManager
 	capacity      int
 
 	mu           sync.Mutex
 	queue        []cmdAndID
 	runningID    runner.RunID
 	runningAbort chan<- struct{}
+
+	startCh  chan cmdAndID
```

Comment:
add comments: 
startCh is used to signal/request starting a command
updateCh is used to signal/request updating the repo